File: feature-dashboard/src/main/java/fr/o80/sample/featuredashboard/presentation/ui/DashboardFragment.java
package fr.o80.sample.featuredashboard.presentation.ui;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.TextView;

import javax.inject.Inject;

import fr.o80.sample.featuredashboard.DashboardService;
import fr.o80.sample.featuredashboard.R;
import fr.o80.sample.lib.core.presenter.Presenter;
import fr.o80.sample.lib.core.ui.BaseFragment;

/**
 * @author Olivier Perez
 */
public class DashboardFragment extends BaseFragment {

    @Inject
    protected DashboardService dashboardService;

    @Override
    public void onResume() {
        super.onResume();
        ((DashboardActivity)getActivity()).component().inject(this);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        ((TextView) view.findViewById(R.id.textView)).setText(R.string.lorem_ipsum);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_dashboard;
    }

    @Override
    protected Presenter presenter() {
        return null;
    }

    @Override
    public void showLoading() {

    }

    @Override
    public void hideLoading() {

    }
}

File: feature-dashboard/src/main/java/fr/o80/sample/featuredashboard/presentation/ui/DashboardFragment.kt
package fr.o80.sample.featuredashboard.presentation.ui

import android.os.Bundle
import fr.o80.sample.featuredashboard.R
import fr.o80.sample.featuredashboard.presentation.presenter.DashboardPresenter
import fr.o80.sample.featuredashboard.presentation.presenter.DashboardView
import fr.o80.sample.lib.core.ui.BaseFragment
import kotlinx.android.synthetic.main.fragment_dashboard.*
import javax.inject.Inject

/**
 * @author Olivier Perez
 */
class DashboardFragment : BaseFragment(), DashboardView {

    @Inject
    lateinit var presenter: DashboardPresenter

    override val layoutId: Int
        get() = R.layout.fragment_dashboard

    override fun presenter(): DashboardPresenter {
        return presenter
    }

    override fun inject() {
        (activity as DashboardActivity).component.inject(this)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        textView.setText(R.string.lorem_ipsum)
    }

}
